modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/" + USER_ID);
+
+        doThrow(new CharonException()).when(userManager).deleteUser(id);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestUpdateWithPUTSuccess")
+    public Object[][] dataToTestUpdateWithPUTSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
 
         SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
         JSONDecoder decoder = new JSONDecoder();
-        return decoder.decodeResource(scimObjectString, schema, new User());
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING, schema, new User());
+        String id = userOld.getId();
+        User userNew = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_UPDATE, schema, new User());
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", null,
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, "userName", "emails",
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, null, "emails",
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+                {id, NEW_USER_SCIM_OBJECT_STRING_UPDATE, null, null,
+                        userNew, userOld, ResponseCodeConstants.CODE_OK},
+        };
     }
 
+    @Test(dataProvider = "dataForTestUpdateWithPUTSuccess")
+    public void testUpdateWithPUTSuccess(String id, String scimObjectString, String
+            attributes, String excludeAttributes, Object scimNewUserObject,
+                                         Object scimOldUserObject, int expectedScimResponseStatus)
+            throws BadRequestException, CharonException, NotImplementedException, NotFoundException {
+
+        User userNew = (User) scimNewUserObject;
+        User userOld = (User) scimOldUserObject;
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        Mockito.when(userManager.getUser(id,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenReturn(userOld);
```

Comment:
can't we merge this into 1 line?

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11

Reply:
is this fixed?

Reply:
Fixed in  commit af77bf94a86e4b649d26ce314d5408b2fde6750d